### PR TITLE
Replace `once_cell` usage with std::sync::OnceLock

### LINKED ISSUE #30: Use once_cell::Lazy instead of std::sync::Once
See summary. Simpler and less easy to get wrong.

### LINKED ISSUE #1141: Replace once_cell with std
Currently, despite requiring 1.70, we are still depending on once_cell. It would be nice to completely remove that dependency and replace it with std where possible.

### PR DESCRIPTION
**Pull request by andy128k**
Closes #30 
Closes #1141 

See also ~~https://github.com/gtk-rs/gir/pull/1532~~ https://github.com/gtk-rs/gir/pull/1537

This change does not fully remove `once_cell`. Full removal will break a contract of `glib::Properties` macro.

### CODE REVIEW

File: glib-macros/src/properties.rs [hunk 1]
```diff
@@ -398,11 +398,10 @@ fn expand_properties_fn(props: &[PropDesc]) -> TokenStream2 {
     quote!(
         fn derived_properties() -> &'static [#crate_ident::ParamSpec] {
```
Comment (sdroege) [inline glib-macros/src/properties.rs:399]: >  Full removal will break a contract of glib::Properties macro.

That's OK though :) Can you update it accordingly?

File: examples/object_subclass/author.rs [hunk 2]
```diff
@@ -23,9 +23,8 @@ mod imp {
     #[glib::derived_properties]
     impl ObjectImpl for Author {
         fn signals() -> &'static [Signal] {
-            static SIGNALS: Lazy<Vec<Signal>> =
```
Comment (sdroege) [inline examples/object_subclass/author.rs:26]: I wonder if we shouldn't wait for stabilization of `LazyLock`. But this also seems simple enough, doesn't really make much difference the way it's used in most places.

@bilelmoussaoui What do you think?

**Review by sdroege — COMMENTED** [review]
Looks good to me otherwise

File: examples/object_subclass/author.rs (hunk 2 above)
Comment (andy128k) [reply examples/object_subclass/author.rs:26]: `LazyLock` will not make it any cleaner or shorter in such cases.

- You need two statements with `OnceLock`: a declaration and an acquiring with initialization.
- You need two statements with `LazyLock`: a declaration with initialization and an acquiring.

OTOH, Initialization of `Regex`es with `LazyLock` would be much cleaner.

File: glib-macros/src/properties.rs (hunk 1 above)
Comment (andy128k) [reply glib-macros/src/properties.rs:399]: Done

**Comment by andy128k** [comment]
I guess, this [PR](https://github.com/gtk-rs/gir/pull/1537) should be merged first.

File: examples/object_subclass/author.rs (hunk 2 above)
Comment (bilelmoussaoui) [reply examples/object_subclass/author.rs:26]: I am fine with this change, at the end of the day, the ideal situation would be having a macro for generating the signals boilerplate.

File: gdk-pixbuf/src/subclass/pixbuf_animation.rs [hunk 3]
```diff
@@ -160,8 +160,10 @@ unsafe extern "C" fn animation_get_size<T: PixbufAnimationImpl>(
     }
 }
 
-static STATIC_IMAGE_QUARK: Lazy<glib::Quark> =
-    Lazy::new(|| glib::Quark::from_str("gtk-rs-subclass-static-image"));
+fn static_image_quark() -> glib::Quark {
```
Comment (bilelmoussaoui) [inline gdk-pixbuf/src/subclass/pixbuf_animation.rs:163]: I am not really a fan of this change tbh, as at the end of the day we don't get rid of once_cell from our dependencies tree. I would keep such Lazy usages

**Review by bilelmoussaoui — APPROVED** [review]
As I mentioned above, not a fan of the GQuark change but no strong opinion. Otherwise lgtm.

File: gdk-pixbuf/src/subclass/pixbuf_animation.rs (hunk 3 above)
Comment (andy128k) [reply gdk-pixbuf/src/subclass/pixbuf_animation.rs:163]: May you explain? This PR literally removes usage of `once_cell` crate.

Comment (bilelmoussaoui) [reply gdk-pixbuf/src/subclass/pixbuf_animation.rs:163]: Never mind, I got confused by the tests/examples deps pulling once_cell.

File: glib-macros/src/properties.rs (hunk 1 above)
Comment (sdroege) [reply glib-macros/src/properties.rs:399]: What contract does that break though? That change is completely internal or not?

**Review by sdroege — APPROVED** [review]
Looks good to me

File: gdk-pixbuf/src/subclass/pixbuf_animation.rs (hunk 3 above)
Comment (bilelmoussaoui) [reply gdk-pixbuf/src/subclass/pixbuf_animation.rs:163]: I actually don't think you need a function here at all, the quark is used only inside a single function which can define the static itself. See https://github.com/gtk-rs/gtk4-rs/pull/1580/commits/3b8e9b04808509278efb14ead45873e4921cf81b

File: glib-macros/src/properties.rs (hunk 1 above)
Comment (andy128k) [reply glib-macros/src/properties.rs:399]: Users of the macro who use `once_cell` will need to change their properties to std cell too.

File: gdk-pixbuf/src/subclass/pixbuf_animation.rs (hunk 3 above)
Comment (andy128k) [reply gdk-pixbuf/src/subclass/pixbuf_animation.rs:163]: Most of the quarks may be moved inside of a function. But there are a few places where they are used across multiple functions. E. g. `PRINT_HANDLER`.

File: glib-macros/src/properties.rs (hunk 1 above)
Comment (bilelmoussaoui) [reply glib-macros/src/properties.rs:399]: That is fine

**Review by bilelmoussaoui — APPROVED** [review]
before landing this, someone should do the same for gstreamer as they use the re-exported once_cell crate from glib iirc

**Comment by sdroege** [comment]
> before landing this, someone should do the same for gstreamer as they use the re-exported once_cell crate from glib iirc

But we have a `Cargo.lock` so I'll take care of that once this is merged :) I'll shortly go over this again and then merge it.

**Review by sdroege — APPROVED** [review]

**Pull request merged**